ec/ecresource/deploymentresource/apm_expanders.go

```diff
@@ -162,3 +160,28 @@ func expandApmConfig(raw interface{}) *models.ApmConfiguration {
 
 	return nil
 }
+
+func defaultApmTopology(topology []*models.ApmTopologyElement) []*models.ApmTopologyElement {
+	for _, t := range topology {
+		if *t.Size.Value > defaultApmSize {
+			t.Size.Value = ec.Int32(defaultApmSize)
+		}
+		if t.ZoneCount > 1 {
+			t.ZoneCount = 1
+		}
+	}
+
+	return topology
+}
+
+func matchApmTopology(id string, topologies []*models.ApmTopologyElement) (*models.ApmTopologyElement, error) {
+	for _, t := range topologies {
+		if t.InstanceConfigurationID == id {
+			return t, nil
+		}
+	}
+	return nil, fmt.Errorf(
+		`apm topology: invalid instance_configuration_id: "%s" doesn't match any of the deployment template instance configurations`,
```

Comment:
Wondering if it's necessary to have this validation as the API validates this as well:
```console
 api error: cluster.invalid_instance_config: For template [I/O Optimized] the plan definition should contain instance configuration [aws.data.highio.i3]. Only [aws.coordinating.m5d] was supplied
```

Reply:
It's not strictly necessary, but it makes handling this case in the code so much easier, since otherwise a `nil` pointer to struct would be returned and that would need to get handled with exactly he same way, do you have any other way in mind?

Reply:
Ah! got it. Nah, it's fine :)